### ISSUE TITLE
Call stack exceeded when getting the length of a circular path

### ISSUE THREAD

**Issue by dbrnz** [svgdotjs/svgdom#35]
The following results in `RangeError: Maximum call stack size exceeded`:

```
const window = require('svgdom')
const document = window.document
const {SVG, registerWindow} = require('@svgdotjs/svg.js')

registerWindow(window , window.document)

const canvas = SVG(document.documentElement)

// A circular path
const path = canvas.path('M0.5 50a49.5 49.5 0 1 0 99 0 49.5 49.5 0 1 0-99 0');

// Gets RangeError: Maximum call stack size exceeded
const len = path.length();
```

**Comment by Fuzzyma** [comment]
I am a bit short on time atm. If you can, try to debug the code with `node file.js - - debug-brk`. You can use your chrome browsers debugger for that purpose

**Comment by dbrnz** [comment]
I've had a quick try at debugging before raising the issue, but am happy to explore more. Could whatever lead to commit e0a21bc be relevant??

**Comment by Fuzzyma** [comment]
Mh, it does not look relevant to me. This was a real bug beeing fixed but it's not like I changed the way how the split function worked. I just fixed the way the new splits are created so you get the correct arcs.
Is splitAt() the method which leads to the callstack error? Worth checking the variables in that case!

**Comment by dbrnz** [comment]
If it helps, the traceback follows; otherwise I'll debug this my tomorrow.
```
/Users/dave/build/mapmaker/node_modules/svgdom/class/Point.js:6
  create: function (x, y) {
                   ^

RangeError: Maximum call stack size exceeded
    at new create (/Users/dave/build/mapmaker/node_modules/svgdom/class/Point.js:6:20)
    at new Arc (/Users/dave/build/mapmaker/node_modules/svgdom/utils/pathUtils.js:180:15)
    at Arc.splitAt (/Users/dave/build/mapmaker/node_modules/svgdom/utils/pathUtils.js:275:7)
    at Arc.length (/Users/dave/build/mapmaker/node_modules/svgdom/utils/pathUtils.js:258:20)
    at Arc.length (/Users/dave/build/mapmaker/node_modules/svgdom/utils/pathUtils.js:266:19)
    at Arc.length (/Users/dave/build/mapmaker/node_modules/svgdom/utils/pathUtils.js:266:19)
    at Arc.length (/Users/dave/build/mapmaker/node_modules/svgdom/utils/pathUtils.js:266:19)
    at Arc.length (/Users/dave/build/mapmaker/node_modules/svgdom/utils/pathUtils.js:266:19)
    at Arc.length (/Users/dave/build/mapmaker/node_modules/svgdom/utils/pathUtils.js:266:19)
    at Arc.length (/Users/dave/build/mapmaker/node_modules/svgdom/utils/pathUtils.js:266:19)
```

**Comment by Fuzzyma** [comment]
Thanks! Don't know if I get anything done until tomorrow so feel free to go ahead :).

**Comment by Fuzzyma** [comment]
So I was able to have a quick look and it seems fahr the splitAt method is really still broken.
The subarc lengths of are alternating in size but they should only decrease. So we are creating wrong arcs in between. Funny enough the alternation approaches a fixed value but that doesn't make it less wrong.

**Comment by dbrnz** [comment]
@Fuzzyma I'm now looking at this and reading up on elliptical arcs. Happy to continue and fix things, although I won't be able to put too much time into it until later in the week.

**Comment by Fuzzyma** [comment]
Cool! It might be possible to fix that differently by not creating 2 new paths but instead sample only points from the main path. So you would sample as much points as you need to achieve the desired accuracy. 

That works because the length measurement is done by calculating the length  of the line from start to endpoint. It doesnt matter if you divide the arc into subarcs or sublines.

**Comment by dbrnz** [comment]
The perils of Javascript..!

**Comment by Fuzzyma** [comment]
Merged and released with 0.0.20. Thank you! :)
